### PR TITLE
DTSBPS-444: Payment repository changes

### PR DESCRIPTION
**Pull request by SomeshRA**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSBPS-444

### Change description ###
Payment repository changes

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/PaymentRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        assertThat(recordCount).isEqualTo(3);
+
+        Optional<List<Payment>> successFullPayments = paymentRepository.findByDocumentControlNumberIn(paymentDcns);
+
+        //Then
+        assertThat(successFullPayments).isPresent();
+        assertThat(successFullPayments.get().size()).isEqualTo(3);
+        assertThat(successFullPayments.get()).as("Successful payment").extracting("status")
+            .contains("REQUESTED", "REQUESTED",
+                             "REQUESTED");
```
Comment (mdayican) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/PaymentRepositoryTest.java:62]: formatting

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/Payment.java [hunk 2]
```diff
@@ -19,6 +20,10 @@
 
     private String documentControlNumber;
 
+    private String status;
+
+    private LocalDateTime lastmodified;
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/Payment.java:25]: should be `Instant` or is LocalDateTime enough ?

Comment (SomeshRA) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/Payment.java:25]: Tests did work with LocalDateTime. If we are using Instant will change it.

Comment (SomeshRA) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/Payment.java:25]: Ok I see we are using Instance will change it.

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/PaymentRepository.java [hunk 3]
```diff
@@ -1,9 +1,20 @@
...
+import org.springframework.data.repository.query.Param;
 
+import java.util.List;
+import java.util.Optional;
 import java.util.UUID;
 
 public interface PaymentRepository extends JpaRepository<Payment, UUID> {
 
+    @Modifying(clearAutomatically = true)
+    @Query("UPDATE Payment p SET p.status = 'REQUESTED'"
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/PaymentRepository.java:15]: `REQUESTED` why ?

Comment (SomeshRA) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/PaymentRepository.java:15]: Payment is requested but it is action-ed sometime later by the PAYHUB

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/PaymentRepository.java:15]: do we have 1 status?

Comment (SomeshRA) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/PaymentRepository.java:15]: yes as of now. Are you expecting any more status?

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/PaymentRepository.java:15]: from our perspective payment is done successfully. we better rename?

Comment (SomeshRA) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/PaymentRepository.java]: Changed to SUBMITTED.

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/PaymentRepositoryTest.java (hunk 1 above)
Comment (SomeshRA) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/PaymentRepositoryTest.java]: Changed to SUBMITTED and formated

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/Payment.java (hunk 2 above)
Comment (SomeshRA) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/Payment.java]: Changed to Instance.

**Review by mdayican — APPROVED** [review]

**Pull request merged**